### PR TITLE
1239 VPN subnet access to a use

### PR DESCRIPTION
**Pull request by ATGE**

### CODE REVIEW

File: SoftLayer/CLI/user/vpn_manual.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+"""List Users."""
```
Comment (allmightyspiff) [inline SoftLayer/CLI/user/vpn_manual.py:1]: Change this doc block to reflect what this command does.

File: SoftLayer/CLI/user/vpn_subnet.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+"""List Users."""
```
Comment (allmightyspiff) [inline SoftLayer/CLI/user/vpn_subnet.py:1]: Change docblock to reflect what this command does.

File: SoftLayer/managers/user.py [hunk 3]
```diff
@@ -267,6 +268,60 @@ def add_api_authentication_key(self, user_id):
...
+
+    def vpn_subnet_add(self, user_id, subnet_ids):
+        """Add subnets for a user.
+
+        :param int user_id: User to edit.
+        :param list subnet_ids: list of subnet Ids.
+        """
+        overrides = [{"userId": user_id, "subnetId": subnet_id} for subnet_id in subnet_ids]
+        return_value = self.override_service.createObjects(overrides)
+        return return_value and self.user_service.updateVpnUser(id=user_id)
```
Comment (allmightyspiff) [inline SoftLayer/managers/user.py:288]: This syntax feels weird to me, and im not sure using AND with a boolean and an object will work great.

Something like this is more clear.
```
        overrides = [{"userId": user_id, "subnetId": subnet_id} for subnet_id in subnet_ids]
        return_value = self.override_service.createObjects(overrides)
        update_success = self.user_service.updateVpnUser(id=user_id)
        if not update_success :
            raise exceptions.SoftLayerAPIError("Overrides created, but unable to update VPN user")
        return return_value 
```

File: SoftLayer/managers/user.py [hunk 4]
```diff
@@ -267,6 +268,60 @@ def add_api_authentication_key(self, user_id):
...
+
+    def vpn_subnet_remove(self, user_id, subnet_ids):
+        """Remove subnets for a user.
+
+        :param int user_id: User to edit.
+        :param list subnet_ids: list of subnet Ids.
+        """
+        overrides = self.get_overrides_list(user_id, subnet_ids)
+        return_value = self.override_service.deleteObjects(overrides)
+        return return_value and self.user_service.updateVpnUser(id=user_id)
```
Comment (allmightyspiff) [inline SoftLayer/managers/user.py:298]: same comment about the return_value as the subnet_add function.

File: tests/managers/user_tests.py [hunk 5]
```diff
@@ -221,3 +220,20 @@ def test_create_user_handle_paas_exception(self):
...
+
+    def test_vpn_manual(self):
+        self.manager.vpn_manual(1234, True)
+        self.assert_called_with('SoftLayer_User_Customer', 'editObject', identifier=1234)
+
+    def test_vpn_subnet_add(self):
+        self.manager.vpn_subnet_add(1234, [1234])
+        self.assert_called_with('SoftLayer_Network_Service_Vpn_Overrides', 'createObjects')
+
+    def test_vpn_subnet_remove(self):
```
Comment (allmightyspiff) [inline tests/managers/user_tests.py:237]: for this and subnet_add tests, make sure the `assert_called_with` is checking the args as well.

Something like this (not sure about the exact ids)
`self.assert_called_with('SoftLayer_Network_Service_Vpn_Overrides', 'deleteObjects', args=({'id',11111, 'subnetId':1234}))`

File: SoftLayer/CLI/user/vpn_subnet.py (hunk 2 above)
Comment (ATGE) [reply SoftLayer/CLI/user/vpn_subnet.py]: fixed

File: SoftLayer/managers/user.py (hunk 3 above)
Comment (ATGE) [reply SoftLayer/managers/user.py]: Syntax updated

File: tests/managers/user_tests.py (hunk 5 above)
Comment (ATGE) [reply tests/managers/user_tests.py:239]: tests updated

**Pull request merged**